sycl/include/sycl/builtins.hpp

```diff
@@ -170,6 +187,100 @@ inline __SYCL_ALWAYS_INLINE
 
 #undef __SYCL_MATH_FUNCTION_2_OVERLOAD_IMPL
 
+#define __SYCL_MATH_FUNCTION_2_SGENFLOAT_Y_OVERLOAD(NAME)                      \
+  template <typename T, size_t N>                                              \
+  inline __SYCL_ALWAYS_INLINE                                                  \
+      std::enable_if_t<detail::is_sgenfloat<T>::value, marray<T, N>>           \
+      NAME(marray<T, N> x, T y) __NOEXC {                                      \
+    marray<T, N> res;                                                          \
+    sycl::vec<T, 2> y_vec{y, y};                                               \
+    for (size_t i = 0; i < N / 2; i++) {                                       \
+      auto partial_res = __sycl_std::__invoke_##NAME<vec<T, 2>>(               \
+          detail::to_vec2(x, i * 2), y_vec);                                   \
+      std::memcpy(&res[i * 2], &partial_res, sizeof(vec<T, 2>));               \
+    }                                                                          \
+    if (N % 2) {                                                               \
+      res[N - 1] = __sycl_std::__invoke_##NAME<T>(x[N - 1], y_vec[0]);         \
+    }                                                                          \
+    return res;                                                                \
+  }
+
+__SYCL_MATH_FUNCTION_2_SGENFLOAT_Y_OVERLOAD(fmax)
+    __SYCL_MATH_FUNCTION_2_SGENFLOAT_Y_OVERLOAD(fmin)
```

Comment:
Formatting seems to be off a bit

Reply:
That's what clang-format suggested😕

Reply:
Yes, clang-format doing this for some reason. Let's see if CI will report this change: https://github.com/intel/llvm/pull/8631/commits/549d30d51f84d949fa5250bbb5e5a6ea6bb49b1e